src/query/api/v1/handler/prometheus/handleroptions/headers.go

```diff
@@ -91,6 +97,51 @@ func AddDBResultResponseHeaders(
 		w.Header().Add(headers.FetchedSeriesCount, fmt.Sprint(meta.FetchedSeriesCount))
 	}
 
+	aggregatedSeriesCount := 0
+	unaggregatedSeriesCount := 0
+	noSamplesCount := 0
+	withSamplesCount := 0
+	for _, m := range meta.MetadataByName {
+		aggregatedSeriesCount += m.Aggregated
+		unaggregatedSeriesCount += m.Unaggregated
+		noSamplesCount += m.NoSamples
+		withSamplesCount += m.WithSamples
+	}
+
+	if aggregatedSeriesCount > 0 {
```

Comment:
Any reason we don't want to add these if it's 0? Just seems like it might lead to headaches dealing with it down the line if they're only sometimes enabled

Reply:
Also perhaps we should gate all these behind a config option; e.g. these headers aren't super useful for anyone running without an aggregator

Reply:
I went with the existing behavior for FetchedSeriesCount as precedent. Gating these behind an option/flag of some sort makes sense.

Reply:
Yeah it could be nice to have this logic disable-able either by config or request header, esp. to skip any of the paths that actually do work such as the fetchProm case where we iterate and check every series.